pkg/api/types.go

```diff
@@ -670,6 +674,19 @@ type DownwardAPIVolumeFile struct {
 	FieldRef ObjectFieldSelector `json:"fieldRef"`
 }
 
+// AzureFile represents an Azure File Service mount on the host that shares a pod's lifetime
+type AzureFileVolumeSource struct {
+	// Azure File Service Account Name
+	AccountName string `json:"accountName"`
+	// the name of Azure File Service Account Key ending in ==
+	KeyName string `json:"keyName"`
```

Comment:
`KeyName` is a very generic and confusing name, and the description is misleading. This field contains the name of the *secret* that contains the credentials for this Azure volume. Name it something like `SecretName` and clarify the comment to indicate it refers to a secret which contains the Azure account name/key (not the actual Azure account key).